gdal/ogr/ogrsf_frmts/cad/libopencad/cadgeometry.cpp

```diff
@@ -0,0 +1,1159 @@
+{
+    extrusion = value;
+}
+
+double CADPoint3D::getXAxisAng() const
+{
+    return xAxisAng;
+}
+
+void CADPoint3D::setXAxisAng( double value )
+{
+    xAxisAng = value;
+}
+
+void CADPoint3D::print() const
+{
+    cout << "|---------Point---------|\n" << "Position: " << "\t" << position.getX() << "\t" << position.getY() <<
+    "\t" << position.getZ() << "\n" << endl;
+}
+
+void CADPoint3D::transform( const Matrix& matrix )
+{
+    position = matrix.multiply( position );
+}
+
+//------------------------------------------------------------------------------
+// CADLine
+//------------------------------------------------------------------------------
+
+CADLine::CADLine()
+{
+    geometryType = CADGeometry::LINE;
+}
+
+CADLine::CADLine( const CADPoint3D& startIn, const CADPoint3D& endIn )
+{
+    geometryType = CADGeometry::LINE;
+    start        = startIn;
+    end          = endIn;
+}
+
+CADPoint3D CADLine::getStart() const
+{
+    return start;
+}
+
+void CADLine::setStart( const CADPoint3D& value )
+{
+    start = value;
+}
+
+CADPoint3D CADLine::getEnd() const
+{
+    return end;
+}
+
+void CADLine::setEnd( const CADPoint3D& value )
+{
+    end = value;
+}
+
+void CADLine::print() const
+{
+    cout << "|---------Line---------|\n" << "Start Position: " << "\t" << start.getPosition().getX() << "\t" <<
+    start.getPosition().getY() << "\t" << start.getPosition().getZ() << "\n" << "End Position: " << "\t" <<
+    end.getPosition().getX() << "\t" << end.getPosition().getY() << "\t" << end.getPosition().getZ() << "\n" << endl;
+}
+
+void CADLine::transform( const Matrix& matrix )
+{
+    start.transform( matrix );
+    end.transform( matrix );
+}
+
+//------------------------------------------------------------------------------
+// CADCircle
+//------------------------------------------------------------------------------
+
+CADCircle::CADCircle() : radius( 0.0f )
+{
+    geometryType = CADGeometry::CIRCLE;
+}
+
+double CADCircle::getRadius() const
+{
+    return radius;
+}
+
+void CADCircle::setRadius( double value )
+{
+    radius = value;
+}
+
+void CADCircle::print() const
+{
+    cout << "|---------Circle---------|\n" << "Position: " << "\t" << position.getX() << "\t" << position.getY() <<
+    "\t" << position.getZ() << "\n" << "Radius: " << radius << "\n" << endl;
+
+}
+
+//------------------------------------------------------------------------------
+// CADArc
+//------------------------------------------------------------------------------
+
+CADArc::CADArc() : CADCircle(), startingAngle( 0.0f ), endingAngle( 0.0f )
+{
+    geometryType = CADGeometry::ARC;
+}
+
+double CADArc::getStartingAngle() const
+{
+    return startingAngle;
+}
+
+void CADArc::setStartingAngle( double value )
+{
+    startingAngle = value;
+}
+
+double CADArc::getEndingAngle() const
+{
+    return endingAngle;
+}
+
+void CADArc::setEndingAngle( double value )
+{
+    endingAngle = value;
+}
+
+void CADArc::print() const
+{
```

Comment:
For the whole library, I think it might be better served by a to_string method that is called by a simple print.

e.g. mirroring the java toString or http://en.cppreference.com/w/cpp/string/basic_string/to_string